EditProfile.java

```diff
@@ -0,0 +1,124 @@
+public class EditProfile implements Initializable {
+    @FXML
+    public TextField name;
+    @FXML
+    public TextField email;
+    @FXML
+    public TextField password;
+    @FXML
+    public TextArea information;
+    User user;
+    private int userId;
+
+    EntityManagerFactory entityManagerFactory = Persistence.createEntityManagerFactory("CourseSystemMng");
+    UserHibControl userHibControl = new UserHibControl(entityManagerFactory);
+
+    @Override
+    public void initialize(URL url, ResourceBundle resourceBundle) {
+    }
+
+    public TextArea getInformation() {
+        return information;
+    }
+
+    public void setInformation(TextArea information) {
+        this.information = information;
+    }
+
+    public void confirmChange(ActionEvent actionEvent) {
+
+        user.setInformation(information.getText());
+
+        if (!checkIfValid(name.getText()) || !isValidPassword(password.getText()) || !isValidEmail(email.getText())) {
+            alertMsg();
+        }
+        else {
+            user.setUsername(name.getText());
+            user.setPassword(password.getText());
+            user.setEmail(email.getText());
+            userHibControl.editUser(user);
+            successAlert();
+        }
+    }
+
+    public void setEditProfile(int userId) {
+            this.userId = userId;
+            User user = userHibControl.getUserById(userId);
+            setData(userId, user);
+    }
+
+    public void setData(int userId, User user) {
+        this.userId = userId;
+        this.user = user;
+        user = userHibControl.getUserById(user.getId());
+        name.setText(user.getUsername());
+        information.setText(user.getInformation());
+        email.setText(user.getEmail());
+        password.setText(user.getPassword());
+    }
```

Comment:
setEditProfile ir setData metodus galima būtų sujungti, nes setEditProfile iškviečia setData metodą, o be to daugiau nieko kito daugiau nedaro